Snowflake.Data.Tests/Mock/MockAzureClient.cs

```diff
@@ -0,0 +1,192 @@
+﻿/*
+ * Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+ */
+
+using Azure;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using Metadata = System.Collections.Generic.IDictionary<string, string>;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using System;
+
+namespace Snowflake.Data.Tests.Mock
+{
+    class MockBlobClient : BlobClient
+    {
+        // Mock Azure data for FileHeader
+        internal const string AzureIV = "MOCK_AZURE_IV";
+        internal const string AzureKey = "MOCK_AZURE_KEY";
+        internal const string AzureMatdesc = "MOCK_AZURE_MATDESC";
+        internal const string SFCDigest = "MOCK_SFC_DIGEST";
+
+        // Mock error message for Azure errors
+        internal const string AzureErrorMessage = "Azure Error Message";
+
+        // Mock content length
+        internal const int ContentLength = 9999;
+
+        // Stores the HttpStatusCode string
+        string _key;
+
+        internal MockBlobClient(string blobName) { _key = MockBlobContainerClient.blobContainerName; }
+
+        internal Exception CreateMockAzureError(string key)
+        {
+            HttpStatusCode statusCode = (HttpStatusCode)Enum.Parse(typeof(HttpStatusCode), key);
+
+            RequestFailedException azureError;
+
+            switch (statusCode)
+            {
+                case HttpStatusCode.BadRequest:
```

Comment:
extract to a method instantiation of an exception and simplify the switch or change to a shorter if (whatever you prefer)